### PR TITLE
Updating backport workflow to include custom branch name

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Updating the backport workflow to resolve permissions issue opensearch-project/OpenSearch#1712 and allow custom branch naming for backport branches to manage branch permissions.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -1,16 +1,20 @@
 name: Backport
 on:
-  pull_request:
```
Comment (ohltyler) [inline .github/workflows/backport.yml:3]: Why the change to `pull_request_target`?

File: .github/workflows/backport.yml [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
+  pull_request_target:
     types:
       - closed
```
Comment (ohltyler) [inline .github/workflows/backport.yml:5]: What happens if, for example, some PR gets labeled as `v1.2.5`, but then gets closed without merging - will this still open a backport PR based off of it?

File: .github/workflows/backport.yml (hunk 1 above)
Comment (VachaShah) [reply .github/workflows/backport.yml:3]: `pull_request` events can only get write permissions if the PR is within the source repo; cross-repo PRs do not get write permissions. We need the write permissions for auto backporting hence `pull_request_target`.

File: .github/workflows/backport.yml (hunk 2 above)
Comment (VachaShah) [reply .github/workflows/backport.yml:5]: The GHA will open PRs only for merged PRs.

**Review by ohltyler — APPROVED** [review]
LGTM!

**Review by amitgalitz — APPROVED** [review]
Thanks for updating this!
